src/rules/mod.rs

```diff
@@ -265,6 +265,7 @@ where
 /// `node`: The parent node to find children token(s) in.
 ///
 /// `token_pred`: A closure returning `true` for all tokens to return.
+#[allow(unused)]
 fn find_descendants_tokens<P>(node: &SyntaxNode, token_pred: P) -> impl Iterator<Item = SyntaxToken>
```

Comment:
Maybe it's better to remove these functions when they are not used. I think they were meant to be general purpose functions to fetch Tokens / SyntaxNode for a given predicate. There exist already a few ways to fetch specific items, e.g. `SyntaxNode::descendants` where the content is also cast into target types.